src/reducers/modal-reducer.js

```diff
@@ -0,0 +1,25 @@
+const modalReducer = (state, { type, payload }) => {
+	switch (type) {
+		case "SET_MODAL":
+			return {
+				...state,
+				message: payload.message,
+				handleConfirm: payload.handleConfirm,
+				handleDismiss: payload.handleDismiss,
+				confirmChoice: payload.confirmChoice,
+				dismissChoice: payload.dismissChoice,
```

Comment:
The key in the payload is the same as the key in the state, so you can just destructed payload to achieve the same result